### PR TITLE
Adding rlimit in spec

### PR DESCRIPTION
**Pull request by rajasec**
Adding rlimit constraints to spec with hard and soft values

Signed-off-by: Rajasekaran <rajasec79@gmail.com>

### CODE REVIEW

File: spec.go [hunk 1]
```diff
@@ -409,6 +420,23 @@ func setupUserNamespace(spec *specs.LinuxSpec, config *configs.Config) error {
...
+		return configs.Rlimit{
+			Type: int(rlimit.Type),
+			Hard: uint64(rlimit.Hard),
+			Soft: uint64(rlimit.Soft),
+		}
+	}
+	for _, rlimit := range spec.Linux.Rlimits {
+		config.Rlimits = append(config.Rlimits, create(rlimit))
+	}
+	return nil
```
Comment (laijs) [inline spec.go]: it always return nil (no error), I think we should remove the return type (error) from this function.

File: spec.go [hunk 2]
```diff
@@ -409,6 +420,23 @@ func setupUserNamespace(spec *specs.LinuxSpec, config *configs.Config) error {
 	return nil
 }
 
+func setupRlimits(spec *specs.LinuxSpec, config *configs.Config) error {
+	if len(spec.Linux.Rlimits) == 0 {
+		return nil
+	}
```
Comment (laijs) [inline spec.go]: it's redundant code

**Comment by rajasec** [comment]
@mrunalp @laijs 
Fixed the comments. Now eliminated the error return type which is not needed for this function.

**Comment by mrunalp** [comment]
@rajasec Can you update the README as well?

**Comment by rajasec** [comment]
@mrunalp 
I've updated README

File: README.md [hunk 3]
```diff
@@ -122,7 +122,13 @@ user with uid and gid of `0` defined within that file-system.
     "linux": {
         "uidMapping": null,
         "gidMapping": null,
-        "rlimits": null,
+         "rlimits": [
```
Comment (mrunalp) [inline README.md]: The indentation seems off here.

**Comment by rajasec** [comment]
@mrunalp 
I've corrected the space (indentation).. Sorry about it.

**Comment by mrunalp** [comment]
LGTM

**Comment by laijs** [comment]
LGTM

**Comment by mrunalp** [comment]
@LK4D4 @crosbymichael ping

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**